test/stress/hammer.go

```diff
@@ -227,90 +259,62 @@ func hammer(ctx context.Context) error {
+		maxActPost := maxPosters / 2
+		operation := "create post (act)"
+		posters := make(chan struct{}, maxActPost)
 		wg.Add(1)
-		go runLooped(ctx, func() error {
-			posters <- struct{}{}
-			t0 := time.Now()
-			next := int(pid.Add(1))
-			defer func() {
-				since := time.Since(t0)
-				var slow string
-				if since > 200*time.Millisecond {
-					slow = " (SLOW)"
-				}
-				h.printf("new post id = %d, took %vms%s", next, float64(since.Microseconds())/1e3, slow)
-			}()
-
-			content := bigData[:rand.Intn(maxContentLen)+1] // random.String(rand.Intn(maxContentLen) + 1) // randomBytes(maxContentLen)
-			h.printf("beginning createPostAsync id = %d, content len = %d (concurrent with %d others)",
-				next, len(content), len(posters)-1)
-			promise, err := h.createPostAsync(ctx, dbid, next, "title_"+strconv.Itoa(next), content)
-			if err != nil {
-				// Continue runLooped if it was a timeout as these are typically
-				// transient and we don't want to let up.
-				if errors.Is(err, context.DeadlineExceeded) {
-					h.printf("Timeout creating post!")
-					err = nil // just keep trying
-				}
-				<-posters
-				return err
-			}
-
-			go func() {
-				timer := time.NewTimer(10 * time.Second)
-				defer timer.Stop()
-				select {
-				case res := <-promise:
-					if err := res.Error(); err != nil {
-						h.printf("createPost failed: %v", err)
-					}
-				case <-timer.C:
-					logger.Error("timed out waiting for create post tx to be mined")
-				}
-
-				<-posters
-			}()
-			return nil
-		}, "create post", postInterval, &logger)
+		go runLooped(ctx,
+			asyncFn(ctx, posters, h.printf, operation,
+				func() (<-chan asyncResp, error) {
+					next := int(pid.Add(1))
+					content := bigData[:contentLen]
+					// content := bigData[:rand.Intn(maxContentLen)+1] // random.String(rand.Intn(maxContentLen) + 1) // randomBytes(maxContentLen)
+					h.printf("beginning createPostAsync id = %d, content len = %d (concurrent with %d others)",
+						next, len(content), len(posters)-1)
+					return asc.createPostAsync(ctx, dbid, next, "title_"+strconv.Itoa(next), content)
+				},
+			),
+			operation, postInterval, &logger,
+		)
 	}
 
-	// Some other things to consider
-
-	// action delete_post($id) public
-	// action delete_user_by_id ($id) public owner
-	// action delete_user() public
-	// action update_user($id, $username, $age) public
-
-	// action list_users() public
-	// action get_user_posts($username) public
-	// action get_user_posts_by_userid($id) public
-	// action multi_select() public
+	// post (procedures)
+	if maxPosters > 0 {
+		maxProcPost := maxPosters / 2
+		operation := "create post (proc)"
+		posters := make(chan struct{}, maxProcPost)
+		wg.Add(1)
+		go runLooped(ctx,
+			asyncFn(ctx, posters, h.printf, operation,
+				func() (<-chan asyncResp, error) {
+					content := bigData[:contentLen]
+					// content := bigData[:rand.Intn(maxContentLen)+1] // random.String(rand.Intn(maxContentLen) + 1) // randomBytes(maxContentLen)
```

Comment:
Maybe add a flag to configure we want fixed size payload or variable?

Reply:
Yeah, good call.

Reply:
Added `-vl` flag for variable length: https://github.com/kwilteam/kwil-db/pull/841/commits/ebf62833893e8d6b23aed85e4e41b43c6f628447